synapse/federation/federation_server.py

```diff
@@ -1092,7 +1092,7 @@ async def _process_incoming_pdus_in_room_inner(
         # has started processing).
         while True:
             async with lock:
-                logger.info("handling received PDU: %s", event)
+                logger.info("handling received PDU in room %s: %s", room_id, event)
```

Comment:
**Question for reviewer**: I also considered updating `FrozenEvent.__repr__` to include the room ID, since it seems a generally useful thing to log about an event. However, I'm wary of making `repr(event)` huge, and normally there are other clues in the logs about which room we are working in, so I decided against that approach. But thoughts on this would be welcome.

Reply:
I think it makes sense to do it this way. If we find ourselves adding more `room_id` logging then we can revisit whether we should fold that into the `__repr__`